File: Scratch Project/Android/MyAudioSong/app/src/main/java/com/example/myaudiosong/MainActivity.java
package com.example.myaudiosong;

import androidx.appcompat.app.AppCompatActivity;

import android.media.MediaPlayer;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {


    private double startTime = 0;
    private double endTime = 0;
    private int forwardTime = 5000;
    private int backwardTime = 5000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

       final Button play = (Button) findViewById(R.id.play_button);
       final Button pause = (Button) findViewById(R.id.pause_button);
       final Button forward = (Button) findViewById(R.id.forward_button);
       final Button backward = (Button) findViewById(R.id.backward_button);

        final MediaPlayer mediaPlayer = MediaPlayer.create(this, R.raw.davissong);
        play.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mediaPlayer.start();
                endTime = mediaPlayer.getDuration();
                startTime = mediaPlayer.getCurrentPosition();
                mediaPlayer.setOnCompletionListener(new MediaPlayer.OnCompletionListener() {
                    @Override
                    public void onCompletion(MediaPlayer mp) {
                        finish();
                        Toast.makeText(getApplicationContext(),"I'm done", Toast.LENGTH_SHORT).show();
                    }
                });
            }
        });

        pause.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mediaPlayer.pause();
                pause.setEnabled(true);
                forward.setEnabled(true);
            }
        });

        forward.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int temp = (int)startTime;
                if((temp + forwardTime)  <= endTime){
                    startTime = startTime + forwardTime;
                    mediaPlayer.seekTo((int)startTime);
                }
            }
        });

        backward.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int temp = (int)startTime;
                if((temp - backwardTime) > 0){
                    startTime = startTime - backwardTime;
                    mediaPlayer.seekTo((int)startTime);
                }
            }
        });



    }
}

File: z01_miscellaneous/belajar03_snackbar/app/src/main/java/id/kotlin/belajar/MainActivity.kt
package id.kotlin.belajar

import android.os.Bundle
import android.widget.Button
import android.widget.LinearLayout
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.snackbar.Snackbar

class MainActivity: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button = findViewById<Button>(R.id.btn_click)
        button.setOnClickListener { Toast.makeText( this, "Hello Toast", Toast.LENGTH_SHORT).show() }

        val rootLayout = findViewById<LinearLayout>(R.id.root_layout)
        val buttonSnack = findViewById<Button>(R.id.btn_click_snack)
        buttonSnack.setOnClickListener { Snackbar.make(rootLayout, "Hello Snack", Snackbar.LENGTH_SHORT).show() }
    }
}